sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -773,36 +773,36 @@ class JDBCSuite extends QueryTest
   }
 
   test("compile filters") {
-    val compileFilter = PrivateMethod[Option[String]](Symbol("compileFilter"))
     def doCompileFilter(f: Filter): String =
-      JDBCRDD invokePrivate compileFilter(f, JdbcDialects.get("jdbc:")) getOrElse("")
+      f.toV2.flatMap(JdbcDialects.get("jdbc:").compileExpression(_)).getOrElse("")
+
     Seq(("col0", "col1"), ("`col0`", "`col1`")).foreach { case(col0, col1) =>
-      assert(doCompileFilter(EqualTo(col0, 3)) === """"col0" = 3""")
-      assert(doCompileFilter(Not(EqualTo(col1, "abc"))) === """(NOT ("col1" = 'abc'))""")
+      assert(doCompileFilter(EqualTo(col0, 3)) === """("col0") = (3)""")
```

Comment:
hmm, why do we wrap the column name and value with `(...)` in v2 predicate?

Reply:
Because the discussion https://github.com/apache/spark/pull/35248#discussion_r801712556